### PR TITLE
Changes to add path existence validation

### PR DESCRIPTION
**Pull request by sr-murthy**

### CODE REVIEW

File: pyxs/exceptions.py [hunk 1]
```diff
@@ -51,6 +51,10 @@ class InvalidPath(ValueError, PyXSError):
     :param bytes path: invalid path value.
 """
 
+class NonexistentPath(ValueError, PyXSError):
+    """
+    Exception raised when a path is not an actual Xenstore path.
```
Comment (superbobry) [inline pyxs/exceptions.py]: I know this is small, but consistency matters. We use XenStore (not the capital S) through out documentation.

File: pyxs/helpers.py [hunk 2]
```diff
@@ -125,6 +126,16 @@ def validate_path(path):
     return path
 
 
+def path_exists(path):
+    """Checks if a given path exists, i.e. is an actual Xenstore path.
+    :exc:`~pyxs.exceptions.InvalidPath` for details
+    """
+    try:
+        return Client().ls(path) != None
```
Comment (superbobry) [inline pyxs/helpers.py]: Are you sure you want it as a function and not as a client method, i.e. `Client.exists`. Also, please use `is not` for comparison with `None`.

File: pyxs/helpers.py [hunk 3]
```diff
@@ -125,6 +126,16 @@ def validate_path(path):
...
+    except PyXSError as e:
+        raise NonexistentPath('%s is not an existing Xenstore path' % path)
```
Comment (superbobry) [inline pyxs/helpers.py]: I think `path` should be the only argument here, because the rest is easily inferred from the exception class name.

**Comment by superbobry** [comment]
Thank you! Feel free to add yourself to the `Contributors` section in `AUTHORS`.

**Comment by sr-murthy** [comment]
Have moved the path checking method to client, and made the function return False in case of a non-existent path rather than raise an exception.

**Comment by sr-murthy** [comment]
Hi,

I have updated the PR - the changes are to make the method name `exists` in `Client`, but I have kept the boolean return in case of a non-existent path because the motivation was to have a simple boolean method for checking whether a given path exists.  So the motivation here is to allow pyxs to have something like `os.path.exists` which allows you to check whether a given file system path exists.  I hope that's OK.

**Comment by superbobry** [comment]
This looks good. Can you add a test for `exists` and then rebase onto the latest `master`?

**Comment by sr-murthy** [comment]
No problem - I see you are using pytest.  I am not familiar with the test workflow.  How do you call the tests?

I forgot to remove the `NonexistentPath` exception class reference from `exceptions.py` in my previous update, so this will removal will go in the next one, along with the test.

**Comment by superbobry** [comment]
> I see you are using pytest. I am not familiar with the test workflow. How do you call the tests?

Simply running `./setup.py test` should work. However, most of the tests need to be executed in Dom0 with root privileges, so I suggest running tests in a separate VM (e.g. VirtualBox).

**Comment by sr-murthy** [comment]
I've used two different machines to work on this, with different GH accounts, and this has become a bit messy.  So I will close this pull request, and re-create a new one with the same objective and content, including the test method.

**Pull request closed without merging**